darts/utils/model_selection.py

```diff
@@ -0,0 +1,218 @@
+from typing import Sequence, Optional, Union, Tuple
+from ..timeseries import TimeSeries
+
+MODEL_AWARE = 'model-aware'
+SIMPLE = 'simple'
+
+
+class SplitTimeSeriesSequence(Sequence):
+    """
+    This class is primarily meant to be instantiated from ``train_test_split()`` function.
+    """
+    def __init__(self,
+                 type: str,
+                 data: Union[TimeSeries, Sequence[TimeSeries]],
+                 test_size: Optional[Union[float, int]] = 0.25,
+                 axis: Optional[int] = 0,
+                 input_size: Optional[int] = None,
+                 horizon: Optional[int] = None,
+                 vertical_split_type: Optional[str] = SIMPLE):
+
+        if type not in ['train', 'test']:
+            raise AttributeError('Value for type parameter should be either `train` or `test`')
+
+        if not data:
+            raise AttributeError('The `data` parameter cannot be empty.')
+
+        if axis not in [0, 1]:
+            raise AttributeError('The `axis` parameter should be either 0 or 1.')
+
+        if vertical_split_type not in [SIMPLE, MODEL_AWARE]:
+            raise AttributeError('`vertical_split_type` can be eiter `simple` or `model-aware`.')
+
+        if axis == 1 and vertical_split_type == MODEL_AWARE and (horizon == 0 or input_size == 0):
+                raise AttributeError("You need to provide non-zero `horizon` and `input_size` parameters when axis=1")
+
+        self.type = type
+        self.data = data
+        self.test_size = test_size
+        self.axis = axis
+        self.input_size = input_size
+        self.horizon = horizon
+        self.vertical_split_type = vertical_split_type
+        self._horizontal_split_index = None
+
+    def _get_horizontal_split_index(self):
+        if not self._horizontal_split_index:
+            if 0 < self.test_size < 1:
+                self._horizontal_split_index = int(len(self.data) * (1 - self.test_size))
+            else:
+                self._horizontal_split_index = len(self.data) - self.test_size
+
+        return self._horizontal_split_index
+
+    def _get_vertical_split_indices(self, ts_length):
+
+        if self.vertical_split_type == SIMPLE:
+            if 0 < self.test_size < 1:
+                test_size = int(ts_length * self.test_size)
+            else:
+                test_size = self.test_size
+
+            test_start_index = ts_length - test_size
+            train_end_index = test_start_index
+
+            if test_start_index < 0:
+                raise AttributeError("`test_size` is bigger then timeseries length")
+
+        else: # model-aware split
+            train_end_index = ts_length - self.horizon
```

Comment:
This means that models might be trained on some parts of the test set. There can even be cases where some whole input/output sub-sequences in the test set also appear in the training set. I guess this is not what we want, right? Because that wouldn't be a "true" test set. 

I would have expected something like `train_end_index = test_start_index + input_size` because you can have one "input" worth of overlap but not one "input + output" of overlap without leakage (unless I'm missing something).